src/NamingStrategy/PathNamingStrategy.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr\NamingStrategy;
+
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Will use the request attributes (hostname, path, headers & body) as filename.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class PathNamingStrategy implements NamingStrategyInterface
+{
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @param array $options available options:
+     *                       - hash_headers:      the list of header names to hash,
+     *                       - hash_body_methods: Methods for which the body will be hashed (Default: PUT, POST, PATCH)
+     */
+    public function __construct(array $options = [])
+    {
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+        $this->options = $resolver->resolve($options);
+    }
+
+    public function name(RequestInterface $request): string
+    {
+        $parts = [$request->getUri()->getHost()];
+
+        $method = strtoupper($request->getMethod());
+
+        $parts[] = $method;
+        $parts[] = str_replace('/', '_', trim($request->getUri()->getPath(), '/'));
+        $parts[] = $this->getHeaderHash($request);
+
+        if ($query = $request->getUri()->getQuery()) {
+            $parts[] = $this->hash($query);
+        }
+
+        if (\in_array($method, $this->options['hash_body_methods'], true)) {
+            $parts[] = $this->hash((string) $request->getBody());
+        }
+
+        return implode('_', array_filter($parts));
+    }
+
+    private function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'hash_headers' => [],
```

Comment:
i was wondering if we should add defaults here (COOKIE, AUTHORIZATION, ACCEPT, ACCEPT-ENCODING, ACCEPT-LANGUAGE) but i guess most of the time you don't care, and when you do you should explicitly configure the headers you need.

Reply:
I think configuration it's up to the user too